classes/package/package_base.php

```diff
@@ -0,0 +1,177 @@
+<?php
+// This file is part of the QuestionPy Moodle plugin - https://questionpy.org
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace qtype_questionpy\package;
+
+use qtype_questionpy\array_converter\array_converter;
+
+/**
+ * Represents a QuestionPy package.
+ *
+ * @package    qtype_questionpy
+ * @copyright  2023 Jan Britz, TU Berlin, innoCampus - www.questionpy.org
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class package_base {
+
+    /**
+     * @var string package shortname
+     */
+    protected $shortname;
+
+    /**
+     * @var string package namespace
+     */
+    protected $namespace;
+
+    /**
+     * @var array package name
+     */
+    protected $name;
+
+    /**
+     * @var string package type
+     */
+    protected $type;
+
+    /**
+     * @var string|null package author
+     */
+    protected $author;
+
+    /**
+     * @var string|null package url
+     */
+    protected $url;
+
+    /**
+     * @var array|null package languages
+     */
+    protected $languages;
+
+    /**
+     * @var array|null package description
+     */
+    protected $description;
+
+    /**
+     * @var string|null package icon
+     */
+    protected $icon;
+
+    /**
+     * @var string|null package license
+     */
+    protected $license;
+
+    /**
+     * @var array|null package tags
+     */
+    protected $tags;
+
+    /**
+     * Constructs package class.
+     *
+     * @param string $shortname
+     * @param string $namespace
+     * @param array $name
+     * @param string $type
+     * @param string|null $author
+     * @param string|null $url
+     * @param array|null $languages
+     * @param array|null $description
+     * @param string|null $icon
+     * @param string|null $license
+     * @param array|null $tags
+     */
+    public function __construct(string $shortname, string $namespace, array $name, string $type,
+                                string $author = null, string $url = null, array $languages = null,
+                                array  $description = null, string $icon = null, string $license = null,
+                                array  $tags = null) {
```

Comment:
```suggestion
    public function __construct(string $shortname, string $namespace, array $name, string $type,
                                ?string $author = null, ?string $url = null, ?array $languages = null,
                                ?array  $description = null, ?string $icon = null, ?string $license = null,
                                ?array  $tags = null) {
```

Bitte auch bei den Properties dann benutzen. Im PHPdoc muss es leider `|null` bleiben, da werden die `?`s nicht verstanden.